src/java.base/share/classes/jdk/internal/foreign/abi/aarch64/CallArranger.java

```diff
@@ -271,7 +287,7 @@ protected void spillStructUnbox(Binding.Builder bindings, MemoryLayout layout) {
             while (offset < layout.byteSize()) {
                 long copy = Math.min(layout.byteSize() - offset, STACK_SLOT_SIZE);
                 VMStorage storage =
-                    storageCalculator.stackAlloc(copy, STACK_SLOT_SIZE);
+                    storageCalculator.stackAlloc(copy, layout.byteAlignment(), false);
```

Comment:
I think maybe passing `layout.byteAlignment` here is a little misleading, as it is unconditionally overwritten to `STACK_SLOT_SIZE` in `stackAlloc` any ways.

I know that's a change I made, but I think it might be better to just use `STACK_SLOT_SIZE` as alignment here, and then I don't think the extra `subSlotPacking` parameter is needed either.

Reply:
> Ok. So it looks like only HFAs are 'packed', and other structs follow the classic approach of passing everything in stack slot sized chunks :/

Yes but I only figured that out by looking at what Clang does. Apple's page at [1] isn't very clear on this.

[1] https://developer.apple.com/documentation/xcode/writing-arm64-code-for-apple-platforms